src/state/component-repository/hub/actions.js

```diff
@@ -0,0 +1,283 @@
+    })
+      .finally(() => {
+        dispatch(toggleLoading(FETCH_REGISTRIES_LOADING_STATE));
+      });
+  })
+);
+
+export const fetchBundleGroups = (url, page = { page: 1, pageSize: 0 }, params = '') => dispatch => (
+  new Promise((resolve) => {
+    getBundleGroups(url, page, params).then((response) => {
+      response.json().then((data) => {
+        if (response.ok) {
+          dispatch(setFetchedBundleGroups(data.payload));
+        } else {
+          dispatch(addErrors(data.errors.map(err => err.message)));
+          data.errors.forEach(err => dispatch(addToast(err.message, TOAST_ERROR)));
+        }
+        resolve();
+      });
+    }).finally(() => {
+    });
+  })
+);
+
+export const fetchBundlesFromRegistryWithFilters = (url, page) => (dispatch, getState) => (
+  new Promise((resolve) => {
+    const state = getState();
+    const filters = getBundleFilters(state);
+    const params = `?${Object.keys(filters).map(k => (filters[k] ? `${k}=${filters[k]}` : ''))}`;
+    dispatch(toggleLoading(FETCH_BUNDLES_LOADING_STATE));
+    getBundlesFromRegistry(url, page, params).then((response) => {
+      response.json().then((data) => {
+        if (response.ok) {
+          dispatch(fetchBundleStatuses(data.payload.map(bundle => bundle.gitRepoAddress)));
+          dispatch(setFetchedBundlesFromRegistry(data.payload));
+          dispatch(setPage(data.metadata));
+        } else {
+          dispatch(addErrors(data.errors.map(err => err.message)));
+          data.errors.forEach(err => dispatch(addToast(err.message, TOAST_ERROR)));
+        }
+        resolve();
+      });
+    }).catch(() => {
+      dispatch(addToast('Failed to fetch bundles from indicated API', TOAST_ERROR));
+    })
+      .finally(() => {
+        dispatch(toggleLoading(FETCH_BUNDLES_LOADING_STATE));
+      });
+  })
+);
+
+export const sendDeleteRegistry = registryId => dispatch => (
+  new Promise((resolve) => {
+    deleteRegistry(registryId).then((response) => {
+      response.json().then((data) => {
+        if (response.ok) {
+          dispatch(addToast(
+            { id: 'app.deleted', values: { type: 'registry', code: data.payload.name } },
+            TOAST_SUCCESS,
+          ));
+          dispatch(fetchRegistries());
+        } else {
+          dispatch(addErrors(data.errors.map(err => err.message)));
+          data.errors.forEach(err => dispatch(addToast(err.message, TOAST_ERROR)));
+        }
+        resolve();
+      });
+    }).finally(() => {
+    });
+  })
+);
+
+export const sendAddRegistry = registryObject => dispatch => (
+  new Promise((resolve) => {
+    addRegistry(registryObject).then((response) => {
+      response.json().then((data) => {
+        if (response.ok) {
+          dispatch(addToast(
+            { id: 'app.created', values: { type: 'registry', code: data.payload.name } },
+            TOAST_SUCCESS,
+          ));
+          dispatch(fetchRegistries());
```

Comment:
I am keen to keep this because also for example when a registry is added, newly added registry has some additional data (for example id) that must be fetched.